api/src/main/java/org/pmiops/workbench/api/DataSetController.java

```diff
@@ -240,15 +244,13 @@ public DataSet apply(DbDataset dataSet) {
         }
       };
 
-  private ConceptSet toClientConceptSet(DbConceptSet conceptSet) {
-    ConceptSet result = ConceptSetsController.toClientConceptSet(conceptSet);
-    if (!conceptSet.getConceptIds().isEmpty()) {
-      Iterable<DbConcept> concepts = conceptDao.findAll(conceptSet.getConceptIds());
-      result.setConcepts(
-          StreamSupport.stream(concepts.spliterator(), false)
-              .map(ConceptsController::toClientConcept)
-              .collect(Collectors.toList()));
-    }
+  private ConceptSet toClientConceptSet(DbConceptSet dbConceptSet) {
+    ConceptSet result = conceptSetMapper.dbModelToClient(dbConceptSet);
+    Iterable<DbConcept> concepts = conceptDao.findAll(dbConceptSet.getConceptIds());
+    result.setConcepts(
+        StreamSupport.stream(concepts.spliterator(), false)
+            .map(ConceptsController::toClientConcept)
+            .collect(Collectors.toList()));
```

Comment:
This looks very similar to `toClientConceptSetWithConcepts`. Could you make a service method that takes in a sort order and unify some of this?

Reply:
Done.